File: app/src/main/java/com/robotnec/budget/app/BudgetApplication.java
package com.robotnec.budget.app;

import android.app.Application;

import com.jakewharton.threetenabp.AndroidThreeTen;
import com.robotnec.budget.app.navigator.AndroidNavigator;
import com.robotnec.budget.core.di.ApplicationGraph;

/**
 * @author zak <zak@swingpulse.com>
 */
public class BudgetApplication extends Application {

    private ApplicationGraph applicationGraph;

    @Override
    public void onCreate() {
        super.onCreate();
        AndroidThreeTen.init(this);
        applicationGraph = new ApplicationGraph(
                new AndroidNavigator(),
                this);
    }

    public ApplicationGraph getApplicationGraph() {
        return applicationGraph;
    }
}

File: app/src/main/java/com/robotnec/budget/app/BudgetApplication.kt
package com.robotnec.budget.app

import android.support.v7.app.AppCompatDelegate
import com.jakewharton.threetenabp.AndroidThreeTen
import com.robotnec.budget.core.di.DaggerApplicationComponent
import dagger.android.support.DaggerApplication

/**
 * @author zak zak@swingpulse.com>
 */
class BudgetApplication : DaggerApplication() {

    public override fun applicationInjector() = DaggerApplicationComponent.builder()
            .application(this)
            .build()

    override fun onCreate() {
        super.onCreate()
        AndroidThreeTen.init(this)
        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
    }
}
